godot-fmt/benches/gdext_tests.rs

```diff
@@ -0,0 +1,34 @@
+use std::{path::PathBuf, str::FromStr};
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use godot_fmt::format_tokens;
+use proc_macro2::TokenStream;
+
+pub fn criterion_benchmark(c: &mut Criterion) {
+    // this little dance is needed because `cargo bench` and `cargo criterion`
+    // set the working directory slightly differently 🫠
+    let current_path = PathBuf::from(file!());
+    let test_cases_dir = current_path
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .join("test-cases");
+
+    for rd in std::fs::read_dir(test_cases_dir).unwrap() {
+        let dir_entry = rd.unwrap();
+        let path = dir_entry.path();
+
+        let name = path.file_stem().unwrap();
+
+        let contents = std::fs::read_to_string(dir_entry.path()).unwrap();
+        let stream = TokenStream::from_str(&contents).unwrap();
+
+        c.bench_function(name.to_str().unwrap(), move |b| {
+            b.iter(|| format_tokens(black_box(stream.clone())))
+        });
+    }

```

Comment:
Some comments:
* `rd` could just be named `file`, then you can do another `let file = file.unwrap();`
* variable `path` can be reused also inside `read_to_string`
* maybe `name` could be declared immediately before use:
  `let name = path.file_stem().unwrap().to_str().unwrap();`

Reply:
I named it `dir_entry` because it's technically not a file 🙈 

Did the other two things, nice catch, thanks!!